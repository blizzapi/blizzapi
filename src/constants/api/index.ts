export { hosts as regionHosts } from "./regionHosts";
