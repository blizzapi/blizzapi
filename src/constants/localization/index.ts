export { defaultLocales } from './defaultLocales';
export { locales } from './locales';
export { regions } from './regions';
export { sc2Realms } from './sc2Realms';
