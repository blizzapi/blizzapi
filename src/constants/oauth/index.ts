import { oAuthHosts } from "./OAuthHosts";
import { RegionIdProperties, RegionId, Uri, OAuthEndpoint } from "../../types";

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> =>
  Object.freeze({
    [RegionId.us]: `${oAuthHosts[RegionId.us]}${endpoint}`,
    [RegionId.eu]: `${oAuthHosts[RegionId.eu]}${endpoint}`,
    [RegionId.kr]: `${oAuthHosts[RegionId.kr]}${endpoint}`,
    [RegionId.cn]: `${oAuthHosts[RegionId.cn]}${endpoint}`,
  });

export const authorizeUris = getRegionPropertyArray(OAuthEndpoint.authorize);
export const tokenUris = getRegionPropertyArray(OAuthEndpoint.token);
export const checkTokenUris = getRegionPropertyArray(OAuthEndpoint.checkToken);
