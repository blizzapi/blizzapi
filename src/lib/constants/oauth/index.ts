import { RegionIdProperties } from '../../../../@interfaces';
import { Uri } from '../../../../@types';
import OAuthHosts from './OAuthHosts';

const getRegionPropertyArray = (endpoint: string): RegionIdProperties<Uri> => ({
  1: `${OAuthHosts[1]}${endpoint}`,
  2: `${OAuthHosts[2]}${endpoint}`,
  3: `${OAuthHosts[3]}${endpoint}`,
  5: `${OAuthHosts[5]}${endpoint}`,
});

const authorizeEndpoint = '/oauth/authorize';
const tokenEndpoint = '/oauth/token';
const checkTokenEndpoint = '/oauth/check_token?token=';

const authorizeUris = getRegionPropertyArray(authorizeEndpoint);
const tokenUris = getRegionPropertyArray(tokenEndpoint);
const checkTokenUris = getRegionPropertyArray(checkTokenEndpoint);

export default {
  authorizeUris: Object.freeze(authorizeUris),
  tokenUris: Object.freeze(tokenUris),
  checkTokenUris: Object.freeze(checkTokenUris),
}
