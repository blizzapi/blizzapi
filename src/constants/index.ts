/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import { ConstantKey } from "../types";
import * as localization from "./localization";
import * as api from "./api";
import * as oauth from "./oauth";

export const constants = {
  [ConstantKey.REGIONS]: localization.regions,
  [ConstantKey.LOCALES]: localization.locales,
  [ConstantKey.DEFAULT_LOCALES]: localization.defaultLocales,
  [ConstantKey.SC2_REALMS]: localization.sc2Realms,
  [ConstantKey.REGION_API_HOSTS]: api.regionHosts,
  [ConstantKey.OAUTH_AUTHORIZE_URIS]: oauth.authorizeUris,
  [ConstantKey.OAUTH_TOKEN_URIS]: oauth.tokenUris,
  [ConstantKey.OAUTH_CHECK_TOKEN_URIS]: oauth.checkTokenUris,
};
