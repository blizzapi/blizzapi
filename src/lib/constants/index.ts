/**
 * Class exposing configuration constants
 *
 * Via:
 *
 * https://develop.battle.net/documentation/guides/using-oauth
 * https://develop.battle.net/documentation/guides/regionality-partitions-and-localization
 */

import * as localization from './localization';
import * as api from './api';
import * as oauth from './oauth';

const CONSTANTS = {
  REGIONS: localization.regions,
  LOCALES: localization.locales,
  DEFAULT_LOCALES: localization.defaultLocales,
  SC2_REALMS: localization.sc2Realms,
  REGION_API_HOSTS: api.regionHosts,
  OAUTH_AUTHORIZE_URIS: oauth.authorizeUris,
  OAUTH_TOKEN_URIS: oauth.tokenUris,
  OAUTH_CHECK_TOKEN_URIS: oauth.checkTokenUris,
};

export default CONSTANTS;
