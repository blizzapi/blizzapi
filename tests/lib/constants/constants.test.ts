import constants from '../../../src/constants';

describe('CONSTANTS class', () => {
  test('should be defined', () => {
    expect(constants).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(constants).toMatchSnapshot();
  });
});
