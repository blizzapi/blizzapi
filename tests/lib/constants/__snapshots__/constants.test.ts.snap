// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CONSTANTS class should match snapshot 1`] = `
Object {
  "DEFAULT_LOCALES": Object {
    "1": 0,
    "2": 0,
    "3": 0,
    "5": 0,
  },
  "LOCALES": Object {
    "1": Array [
      "en_US",
      "es_MX",
      "pt_BR",
    ],
    "2": Array [
      "en_GB",
      "es_ES",
      "fr_FR",
      "ru_RU",
      "de_DE",
      "pt_PT",
      "it_IT",
    ],
    "3": Array [
      "ko_KR",
      "zh_TW",
    ],
    "5": Array [
      "zh_CN",
    ],
  },
  "OAUTH_AUTHORIZE_URIS": Object {
    "1": "https://us.battle.net/oauth/authorize",
    "2": "https://eu.battle.net/oauth/authorize",
    "3": "https://apac.battle.net/oauth/authorize",
    "5": "https://www.battlenet.com.cn/oauth/authorize",
  },
  "OAUTH_CHECK_TOKEN_URIS": Object {
    "1": "https://us.battle.net/oauth/check_token?token=",
    "2": "https://eu.battle.net/oauth/check_token?token=",
    "3": "https://apac.battle.net/oauth/check_token?token=",
    "5": "https://www.battlenet.com.cn/oauth/check_token?token=",
  },
  "OAUTH_TOKEN_URIS": Object {
    "1": "https://us.battle.net/oauth/token",
    "2": "https://eu.battle.net/oauth/token",
    "3": "https://apac.battle.net/oauth/token",
    "5": "https://www.battlenet.com.cn/oauth/token",
  },
  "REGIONS": Object {
    "1": Array [
      "us",
    ],
    "2": Array [
      "eu",
    ],
    "3": Array [
      "kr",
      "tw",
    ],
    "5": Array [
      "cn",
    ],
  },
  "REGION_API_HOSTS": Object {
    "1": "https://us.api.blizzard.com",
    "2": "https://eu.api.blizzard.com",
    "3": Array [
      "https://kr.api.blizzard.com",
      "https://tw.api.blizzard.com",
    ],
    "5": "https://gateway.battlenet.com.cn",
  },
  "SC2_REALMS": Object {
    "1": Array [
      1,
      2,
    ],
    "2": Array [
      1,
      2,
    ],
    "3": Array [
      1,
      2,
    ],
    "5": Array [
      1,
    ],
  },
}
`;
